src/default-directory-provider.coffee

```diff
@@ -0,0 +1,42 @@
+{Directory} = require 'pathwatcher'
+fs = require 'fs-plus'
+path = require 'path'
+
+module.exports =
+class DefaultDirectoryProvider
+
+  # Public: Create a Directory that corresponds to the specified URI.
+  #
+  # * `uri` {String} The path to the directory to add.
+  # * `existingPaths` {Array} The existing paths in atom.project.
+  # Returns:
+  # * {Directory} if the given URI is compatible with this provider.
+  # * {string} if the uri maps to a path an existingPath, in which case
+  #     the path in existingPaths is returned.
```

Comment:
I think this `String` return value makes the method interface a little too complex. What's the use case that you're thinking of where this uniqueness check needs to defined by the provider, rather than just being done inside the project? Maybe we can solve the problem a different way.